internal/node/node_loader.js

```diff
@@ -71,408 +74,11 @@ node_loader: running ${TARGET} with
-
-  return { runfilesManifest, reverseRunfilesManifest, binRoot, genRoot, localWorkspacePath };
-}
-const { runfilesManifest, reverseRunfilesManifest, binRoot, genRoot, localWorkspacePath } =
-    // On Windows, Bazel sets RUNFILES_MANIFEST_ONLY=1.
-    // On every platform, Bazel also sets RUNFILES_MANIFEST_FILE, but on Linux
-    // and macOS it's faster to use the symlinks in RUNFILES_DIR rather than resolve
-    // through the indirection of the manifest file.
-    // We also need to construct a reverse map to resolve relative files from existing
-    // manifest entries.
-    process.env.RUNFILES_MANIFEST_ONLY === '1' &&
-    loadRunfilesManifest(process.env.RUNFILES_MANIFEST_FILE);
-
-function isFile(res) {
-  try {
-    return fs.statSync(res).isFile();
-  } catch (e) {
-    return false;
-  }
-}
-
-function isDirectory(res) {
-  try {
-    return fs.statSync(res).isDirectory();
-  } catch (e) {
-    return false;
-  }
-}
-
-function readDir(dir) {
-  return fs.statSync(dir).isDirectory() ?
-      Array.prototype.concat(...fs.readdirSync(dir).map(f => readDir(path.join(dir, f)))) :
-      dir.replace(/\\/g, '/');
-}
-
-function loadAsFileSync(res) {
-  if (isFile(res)) {
-    return res;
-  }
-  if (isFile(res + '.js')) {
-    return res;
-  }
-  return null;
-}
-
-function loadAsDirectorySync(res) {
-  const pkgfile = path.join(res, 'package.json');
-  if (isFile(pkgfile)) {
-    try {
-      const pkg = JSON.parse(fs.readFileSync(pkgfile, 'UTF-8'));
-      const main = pkg['main'];
-      if (main) {
-        if (main === '.' || main === './') {
-          main = 'index';
-        }
-
-        let maybe = loadAsFileSync(path.resolve(res, main));
-        if (maybe) {
-          return maybe;
-        }
-
-        maybe = loadAsDirectorySync(path.resolve(res, main));
-        if (maybe) {
-          return maybe;
-        }
-      }
-    } catch (e) {
-    }
-  }
-  return loadAsFileSync(path.resolve(res, 'index'));
-}
-
-function resolveManifestFile(res) {
-  const maybe = runfilesManifest[res] || runfilesManifest[res + '.js'];
-  if (maybe) {
-    return maybe;
-  }
-  // Look for tree artifacts that match and update
-  // the runfiles with files that are in the tree artifact.
-  // Attempt to resolve again with the updated runfiles
-  // if a tree artifact matched.
-  let segments = res.split('/');
-  segments.pop();
-  while (segments.length) {
-    const test = segments.join('/');
-    const tree = runfilesManifest[test];
-    if (tree && isDirectory(tree)) {
-      // We have a tree artifact that matches
-      const files = readDir(tree).map(f => path.relative(tree, f).replace(/\\/g, '/'));
-      files.forEach(f => {
-        runfilesManifest[path.posix.join(test, f)] = path.posix.join(tree, f);
-      })
-      return runfilesManifest[res] || runfilesManifest[res + '.js'];
-    }
-    segments.pop();
-  }
-}
+// we patch require here
+module.constructor._load(patchModulePath)
```

Comment:
why this not just the provided `require`? 
it's not patched yet and if it was you would probably want to honor changes other loaders may have made.
I can think of a few cases where you might need to work around other patches to the module system but i'd like to know which =)

Reply:
Yeah, I tried this with `require` and and for some reason it just wouldn't work, no error or anything. But that shouldn't matter too much since this isn't what we want to go with anyway, was just testing some ideas

Reply:
replaced with the `--require` implementation